tachyon/crypto/commitments/merkle_tree/field_merkle_tree/field_merkle_tree.h

```diff
@@ -45,16 +45,15 @@ class FieldMerkleTree {
 
   template <typename Hasher, typename PackedHasher, typename Compressor,
             typename PackedCompressor>
-  static FieldMerkleTree Build(const Hasher& hasher,
-                               const PackedHasher& packed_hasher,
-                               const Compressor& compressor,
-                               const PackedCompressor& packed_compressor,
-                               std::vector<math::RowMajorMatrix<F>>&& leaves) {
+  static FieldMerkleTree Build(
+      const Hasher& hasher, const PackedHasher& packed_hasher,
+      const Compressor& compressor, const PackedCompressor& packed_compressor,
+      std::vector<Eigen::Map<math::RowMajorMatrix<F>>>&& leaves) {
     TRACE_EVENT("Utils", "FieldMerkleTree::Build");
     CHECK(!leaves.empty());
 
-    std::vector<RowMajorMatrixView> sorted_leaves =
-        base::Map(leaves, [](const math::RowMajorMatrix<F>& matrix) {
+    std::vector<RowMajorMatrixView> sorted_leaves = base::Map(
+        leaves, [](const Eigen::Map<math::RowMajorMatrix<F>>& matrix) {
```

Comment:
If `Eigen::Map` has only pointer(or reference), it needs to be copy to pass to lambda function.
```suggestion
        leaves, [](Eigen::Map<math::RowMajorMatrix<F>> matrix) {
```

Reply:
For the value whose size is less than or equal to 16 byte, we concluded that we should copy. But `sizeof(Eigen::Map<math::RowMajorMatrix<F>>` results in 24 byte, and I don't know if it's good to pass by value or by reference...